server/src/main/java/org/elasticsearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -237,23 +277,53 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
             return;
         }
 
-        Boolean value = context.parseExternalValue(Boolean.class);
-        if (value == null) {
-            XContentParser.Token token = context.parser().currentToken();
-            if (token == XContentParser.Token.VALUE_NULL) {
-                if (fieldType().nullValue() != null) {
-                    value = fieldType().nullValue();
+        Boolean value = null;
+        try {
+            value = context.parseExternalValue(Boolean.class);
+            if (value == null) {
+                final XContentParser parser = context.parser();
+                XContentParser.Token token = parser.currentToken();
+                if (token == XContentParser.Token.VALUE_NULL) {
+                    if (fieldType().nullValue() != null) {
+                        value = fieldType().nullValue();
+                    }
+                } else {
+                    if (indexCreatedVersion.onOrAfter(Version.V_6_0_0_alpha1)) {
```

Comment:
Wondering what the general goal is for backwards-compatiblity here? Should this change got to some 6.x branch? If yes, I think its version should be used here, otherwise probably V_7_0_0?

Reply:
When i grabbed master the other day, this entire block was present with all the version checking and logging.

When i rebased earlier today, the legacy check/handling was removed.

Should i remove the legacy code and my associated tests ? Just wondering if the legacy should be remove entirely ?

Your comments for lines above/below with this block are all legacy code which i have kept. I only added the try/catch and skipChildren bit to fix "skipping" over the object/array so the parser would be the proper spot to continue parsing the next field.